scripts/globals/pets/avatar.lua

```diff
@@ -0,0 +1,510 @@
+        actorWeather == xi.combat.element.strongDoubleWeather[petElement]
+    then
+        castingCooldown = castingCooldown - 2
+    -- Weak weathers.
+    elseif
+        actorWeather == xi.combat.element.weakSingleWeather[petElement] or
+        actorWeather == xi.combat.element.weakDoubleWeather[petElement]
+    then
+        castingCooldown = castingCooldown + 2
+    end
+
+    local dayElement = VanadielDayElement()
+    -- Strong day.
+    if dayElement == petElement then
+        castingCooldown = castingCooldown - 3
+    -- Weak day.
+    elseif dayElement == xi.combat.element.weakDay[petElement] then
+        castingCooldown = castingCooldown + 3
+    end
+
+    -- "Buff mode" is enabled for first cast after spawn and after casting any enhancing magic
+    if pet:getLocalVar(buffModeVar) ~= 1 then
+        castingCooldown = math.floor(castingCooldown / 2)
+    end
+
+    -- cast delay is ~1s past the finish of last spell, so we add casting time (or time since spell interrupt) to the castingCooldown
+    -- this is done by simply tracking the elapsed time since action is no longer xi.action.MAGIC_CASTING
+    local lastCastTime = pet:getLocalVar(lastCastTimeVar)
+    local lastCastTimeStamp = pet:getLocalVar(lastCastTimeStampVar)
+    if
+        lastCastTimeStamp > 0 and
+        os.time() - lastCastTimeStamp > lastCastTime
+    then
+        lastCastTime = os.time() - lastCastTimeStamp
+    end
+
+    if
+        lastCastTime > 0 and
+        pet:getCurrentAction() ~= xi.action.MAGIC_CASTING
+    then
+        pet:setLocalVar(lastCastTimeStampVar, 0)
+        pet:setLocalVar(lastCastTimeVar, lastCastTime)
+    end
+
+    pet:setMobMod(xi.mobMod.MAGIC_COOL, lastCastTime + castingCooldown)
+end
+
+xi.pets.avatar.onMobSpawn = function(pet)
+    local master = pet:getMaster()
+    if
+        not master or
+        master:getObjType() ~= xi.objType.PC
+    then
+        return
+    end
+
+    -- add listener to player to fine-tune spirit pact cast delays in realtime
+    if
+        pet:getPetID() <= xi.petId.DARK_SPIRIT
+    then
+        -- stops the pet from immediately casting a spell on spawn and respecting the cooldowns by exiting early if MAGIC_COOL is 1
+        pet:setMobMod(xi.mobMod.MAGIC_COOL, 1)
+        pet:setMod(xi.mod.MPP, 500)
+        pet:updateHealth()
+        pet:setMP(pet:getMaxMP())
+
+        master:addListener('TICK', playerListenerVar, function(playerArg)
+            local petArg = playerArg:getPet()
+
+            if petArg and petArg:getMobMod(xi.mobMod.MAGIC_COOL) > 1 then
+                setMagicCastCooldown(petArg)
+            end
+        end)
+
+        master:addListener('ABILITY_USE', playerListenerVar .. 'ABILITY', function(playerArg, target, ability, action)
+            local petArg = playerArg:getPet()
+            local abilityName = string.lower(ability:getName())
+
+            if
+                petArg and
+                (abilityName == 'assault' or
```

Comment:
Can we not use the ability IDs here? we typically don't use strings for this sort of thing

Reply:
ah we certainly can, i'm still in the old mindset of not having enums for _everything_